contracts/HydraChain/HydraChain.sol

```diff
@@ -127,17 +127,45 @@ contract HydraChain is
      * @return Returns true if the validator is subject to a ban
      */
     function isSubjectToBan(address validator) public view override returns (bool) {
-        uint256 lastCommittedEndBlock = epochs[currentEpochId - 1].endBlock;
-        // check if the threshold is reached when the method is not executed by the owner (governance)
-        if (msg.sender != owner() && lastCommittedEndBlock - validatorsParticipation[validator] < banThreshold) {
-            return false;
+        // check if the owner (governance) is calling or the validator is already subject to ban on previous exit
+        if (msg.sender == owner() || isSubjectToBanAfterExit[validator]) {
+            return true;
         }
 
-        return true;
+        uint256 lastCommittedEndBlock = _commitBlockNumbers[currentEpochId - 1];
+        uint256 valiatorParticipation = validatorsParticipation[validator];
+        // check if the validator is active and the last participation is less than the threshold
+        if (
+            validators[validator].status == ValidatorStatus.Active &&
+            lastCommittedEndBlock > valiatorParticipation &&
```

Comment:
I don't think the opposite is possible

Reply:
When we activate validators we set the valiatorParticipation to the current block